### PR TITLE
Story#646

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: controllers/v1/entities.js [hunk 1]
```diff
@@ -441,6 +441,66 @@ module.exports = class Entities extends Abstract {
       }
     });
   }
+
+    /**
+  * @api {get} /kendra/api/v1/entities/getUsersByEntityAndRole/:entityId?role=""
+  * @apiVersion 1.0.0
+  * @apiName Get users by entityId and role.
+  * @apiGroup Entities
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
```
Comment (aks30) [inline controllers/v1/entities.js]: @deepakr123 - Make it internal access token only.

File: module/entities/helper.js [hunk 2]
```diff
@@ -660,4 +661,105 @@ module.exports = class EntitiesHelper {
...
+                },
+                "_source":  [`data.roles.${role}`,"data.externalId"]
+              }
+
+            let entities = await elasticSearch.searchDocumentFromIndex
+            (
+                process.env.ELASTICSEARCH_ENTITIES_INDEX,
+                "_doc",
+                queryObject,
+                null,
```
Comment (aks30) [inline module/entities/helper.js]: @deepakr123 - Pass it as "all"

Comment (deepakr123) [reply module/entities/helper.js]: resolved

File: controllers/v1/entities.js (hunk 1 above)
Comment (deepakr123) [reply controllers/v1/entities.js]: resolved

**Pull request merged**